examples/c++/risc0/example.cu

```diff
@@ -0,0 +1,284 @@
+
+#include <iostream>
+#include <memory>
+#include <vector>
+#include <list>
+
+#include "polynomials/polynomials.h"
+#include "polynomials/cuda_backend/polynomial_cuda_backend.cuh"
+#include "ntt/ntt.cuh"
+
+using namespace polynomials;
+
+// define the polynomial type
+typedef Polynomial<scalar_t> Polynomial_t;
+
+// RISC-V register type
+typedef int64_t rv_t;
+
+// Convert RISC-V registers to Finite Fields
+void to_ff(rv_t* rv, scalar_t* s, size_t n) {
+  for (int i = 0; i < n; ++i) {
+    s[i] = scalar_t::from(rv[i]);
+  }
+}
+
+void p_print(Polynomial_t * p, int logn, scalar_t shift, std::string header = "Print Vector") {
+  std::cout << header << std::endl;
+  auto n = 1 << logn;
+  auto omega = scalar_t::omega(logn);
+  auto x = shift;
+  for (int i = 0; i < n; ++i) {
+    std::cout << i << ": " << (*p)(x) << std::endl;
+    x = x*omega;
+  }
+}
+
+// value to polynomial
+Polynomial_t p_value(scalar_t value) {
+  auto coeff = std::make_unique<scalar_t[]>(1);
+  coeff[0] = value;
+  auto p_value = Polynomial_t::from_coefficients(coeff.get() , 1);
+  return p_value;
+}
+
+Polynomial_t p_rotate(Polynomial_t* p, int logn) {
+  auto n = 1 << logn;
+  auto evaluations_rou_domain = std::make_unique<scalar_t[]>(n);
+  // TBD: rebase to use evaluate_on_rou_domain
+  // p->evaluate_on_rou_domain(logn, evaluations_rou_domain);
+  auto omega = scalar_t::omega(logn);
+  auto x = scalar_t::one();
+  for (int i = 0; i < n; ++i) {
+    evaluations_rou_domain[i] = (*p)(x);
+    x = x*omega;
+  }
+  scalar_t tmp  = evaluations_rou_domain[n-1];
+  for (int i = n-1; i > 0; --i) {
+    evaluations_rou_domain[i] = evaluations_rou_domain[i-1];
```

Comment:
Maybe this can be computed via coset ntt by evaluate_on_rou_domain and avoiding this copy loop.

Reply:
NTT would be n*logn, copy is n. I'd keep as is.

Reply:
You do the ntt anyway. I meant that you can avoid the O(n) copy and make it O(1), since you shift by one element